flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/window/groupwindow/triggers/ProcessingTimeTriggers.java

```diff
@@ -109,10 +109,7 @@ public boolean onProcessingTime(long time, W window) throws Exception {
             ReducingState<Long> nextFiring = ctx.getPartitionedState(nextFiringStateDesc);
             Long timer = nextFiring.get();
             if (timer != null && timer == time) {
-                long newTimer = time + interval;
-                ctx.registerProcessingTimeTimer(newTimer);
```

Comment:
I am curious about the reasoning why this  ctx.registerProcessingTimeTimer(newTimer); can be removed. The effect would appear to be that instead of a chain of timers, there will now will just be one for the window. The  comments in this file say " Trigger every a given interval, the first trigger time is interval after the first element in
the pane." . The first trigger time implies there will be subsequent ones. 

I am not sure how this change is scoped to the early/late firing scenarios mentioned in the PR conversation.